crates/contract/src/call.rs

```diff
@@ -190,36 +189,36 @@ impl CallDecoder for () {
 /// [sol]: alloy_sol_types::sol
 #[derive(Clone)]
 #[must_use = "call builders do nothing unless you `.call`, `.send`, or `.await` them"]
-pub struct CallBuilder<P, D> {
-    // TODO: this will not work with `send_transaction` and does not differentiate between EIP-1559
-    // and legacy tx
-    request: TransactionRequest,
+pub struct CallBuilder<N: Network, T, P, D> {
+    request: N::TransactionRequest,
     block: Option<BlockId>,
     state: Option<StateOverride>,
     provider: P,
     decoder: D,
+    transport: PhantomData<T>,
 }
 
 // See [`ContractInstance`].
-impl<P: TempProvider> DynCallBuilder<P> {
+impl<N: Network, T: Transport + Clone, P: Provider<N, T>> DynCallBuilder<N, T, P> {
```

Comment:
Confirming the `Clone` bound is OK for Transport?

Reply:
it's necessary for the current implementation, and it can't be a supertrait because it would break object safety